### PR TITLE
Update mentions of sphinx-apidoc

### PR DESCRIPTION
**Pull request by amishatishpatel**
Specifically, to be more informative/useful about its usage.

Also tweaked the dev-guide to discuss contributions/Pull Requests more explicitly.

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in `doc/`
- [x] (N/A) Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-1159.

### CODE REVIEW

File: doc/dev-guide.rst [hunk 1]
```diff
@@ -143,3 +134,37 @@ The module contains the following boiler-plate files:
...
+.. code-block:: bash
+
+  user@host:~/katgpucbf$ sphinx-apidoc -efo doc/ src/
+
+.. note::
+
+    The above command will likely generate a :file:`modules.rst` file, which is
+    not necessary to commit. Furthermore, if you aren't comfortable generating
+    the documents straight away, you can append a ``--dry-run`` to the above
+    command to see what files will be generated.
```
Comment (amishatishpatel) [inline doc/dev-guide.rst:170]: I was toying with splitting these two sentences into a `note::` and ` tip::`, but they're relatively short and looked strange when separated.

File: doc/dev-guide.rst [hunk 2]
```diff
@@ -35,7 +35,7 @@ work, you can just source the virtual environment directly:
 
 .. code-block:: bash
 
-  source .venv/bin/activate
+  user@host:~/katgpucbf$ source .venv/bin/activate
```
Comment (james-smith-za) [inline doc/dev-guide.rst:38]: I don't really think this is necessary. The code block was sufficient.

File: doc/dev-guide.rst (hunk 1 above)
Comment (james-smith-za) [reply doc/dev-guide.rst:170]: I'd remove from "Furthermore" onwards. Git exists for a reason, so if you mess stuff up you can restore.

File: doc/dev-guide.rst [hunk 3]
```diff
@@ -143,3 +134,37 @@ The module contains the following boiler-plate files:
...
+
+:mod:`katgpucbf` holds documentation within its code-base.
+:external+sphinx-apidoc:doc:`sphinx-apidoc <index>` provides a manner to generate
+module documentation as reStructuredText. If you, the developer, add or remove
+a module or file, execute the full ``sphinx-apidoc`` command below to regenerate
+the module documentation with your updates.
+
+.. code-block:: bash
+
+  user@host:~/katgpucbf$ sphinx-apidoc -efo doc/ src/
```
Comment (james-smith-za) [inline doc/dev-guide.rst:163]:
```suggestion
sphinx-apidoc -efo doc/ src/
```

If you really insist, the `$` at the beginning is sufficient.

**Review by james-smith-za — CHANGES_REQUESTED** [review]
Overall I'm happy, these are just nit-picks.

File: doc/dev-guide.rst (hunk 2 above)
Comment (amishatishpatel) [reply doc/dev-guide.rst:38]: I included it here (and below) to more clearly imply/show without explicitly saying where the commands should be run from (i.e. root katgpucbf directory).

Comment (amishatishpatel) [reply doc/dev-guide.rst:38]: Too right, too right. Will roll back and update accordingly.

**Review by james-smith-za — APPROVED** [review]
LGTM.

**Pull request merged**